lib/console_web/controllers/router/device_controller.ex

```diff
@@ -79,33 +80,52 @@ defmodule ConsoleWeb.Router.DeviceController do
         cf_list_enabled = config_profile_to_use.cf_list_enabled
         rx_delay = config_profile_to_use.rx_delay
 
-        multi_buy_value =
-          case length(device.labels) do
-            0 ->
-              case device.multi_buy_id do
-                nil -> 1
-                _ -> if device.multi_buy.value == 21, do: 9999, else: device.multi_buy.value
-              end
-            _ ->
-              label_multi_buys =
-                device.labels
-                |> Enum.map(fn l -> l.multi_buy end)
-                |> Enum.filter(fn mb -> mb != nil end)
-                |> Enum.map(fn mb -> mb.value end)
-
-              case length(label_multi_buys) do
+        packet_config_to_use =
+          case device.packet_config do
+            nil ->
+              labels_with_packet_configs = Enum.filter(device.labels, fn l -> l.packet_config_id != nil end)
+              case length(labels_with_packet_configs) do
                 0 ->
-                  case device.multi_buy_id do
-                    nil -> 1
-                    _ -> if device.multi_buy.value == 21, do: 9999, else: device.multi_buy.value
-                  end
+                  %{
+                    multi_buy_value: 1,
+                    preferred_hotspots: []
+                  }
                 _ ->
-                  max_value = Enum.max(label_multi_buys)
-                  case device.multi_buy_id do
-                    nil -> if max_value == 21, do: 9999, else: max_value
-                    _ -> if device.multi_buy.value == 21, do: 9999, else: device.multi_buy.value
+                  device_label = Labels.get_latest_applied_device_label(device.id, Enum.map(labels_with_packet_configs, fn l -> l.id end))
```

Comment:
this means the latest label will apply, but if the latest label doesnt have any packet settings, other labels with packet settings will be ignored, which isnt what we want probably.

Reply:
I am looking within `labels_with_config_profiles` which all have a config profile. from those, i'm pulling the latest.